apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -286,19 +287,26 @@ impl ShellMode {
     ///
     /// These events can then be queried from this node's `/events` RPC
     /// endpoint, until they are ejected fron the event log some time later.
+    ///
     /// Ejecting events from a node's log happens after they have expired,
     /// which is configured on a per-node basis based on a specific number
-    /// of block heights parameter.
-    pub fn log_events(&self, events: Vec<Event>) {
+    /// of block heights parameter. This is the reason for including a
+    /// `block_height` parameter in this call.
+    pub fn log_events(&self, block_height: BlockHeight, events: Vec<Event>) {
```

Comment:
What type of places will this function be called from? I'm guessing end of `FinalizeBlock`? It would be cleaner to return a `Result`

Reply:
(I just saw the docstring for `EventLog` mentions `FinalizeBlock`)

Reply:
yeah from `FinalizeBlock`. idk, `Shell::broadcast` does the same. besides, we wouldn't be able to recover from the `Result` anyway...

Reply:
Yes, if this fails, the node should shut down because of catastophic failure.

Reply:
If the issue here to not return the `Result` is that we should terminate the program instead, we should do this in a more ordered way than panicking. We should propagate an error to the caller which may then be responsible for shutting down, or passing the error along further. 

This fn shouldn't be responsible for either forwarding events or potentially shutting down the ledger - we should centralize this responsibility somewhere. Right now though it's spread out across multiple places and it would be clearer if we start returning errors where they can occur, even ones that should cause the process to terminate, if this makes sense.

Reply:
for that to be possible, all `Shell` ABCI methods would need to return a `Result`. or at least the ones we knew for sure would have failing ops, like `FinalizeBlock`, to send events to the event log.